lib/response.ex

```diff
@@ -1,56 +1,34 @@
 defmodule HTTPServer.Response do
+  alias HTTPServer.Response.Headers
+  require Logger
   defstruct status_code: nil, status_message: "", resource: nil, headers: %{}, body: ""
 
   @type t :: %__MODULE__{
           status_code: 200 | 204 | 404,
           status_message: String.t(),
           resource: String.t(),
-          headers: %{optional(String.t()) => any},
+          headers: %Headers{},
           body: String.t()
         }
 
   @carriage_return "\r\n"
-  @host "0.0.0.0:4000"
 
-  def send_resp(status_code, body \\ "", headers \\ %{}) do
-    %__MODULE__{
+  def send_resp(status_code, body, headers) do
+    Logger.info("HEADERS -> #{headers}")
+
+    response = %__MODULE__{
       status_code: status_code,
       status_message: status_message(status_code),
       resource: "HTTP/1.1",
       headers: headers,
       body: body
     }
-  end
-
-  def build_headers(body) do
-    %{
-      "Content-Length" => "#{String.length(body)}",
-      "Content-Type" => "text/plain",
-      "Host" => @host
-    }
-  end
-
-  def build_headers(body, headers) do
-    Map.merge(build_headers(body), headers)
-  end
 
-  def build_allow_header(methods) do
-    methods =
-      if Enum.member?(methods, "GET") do
-        methods ++ ["HEAD", "OPTIONS"]
-      else
-        methods ++ ["OPTIONS"]
-      end
+    non_nil_headers =
+      for {k, v} <- Map.from_struct(response.headers), v != nil, into: %{}, do: {k, v}
 
-    %{
-      "Allow" => "#{Enum.join(methods, ", ")}"
-    }
-  end
-
-  def build_location_header(path) do
-    %{
-      "Location" => "http://#{@host}#{path}"
-    }
+    Logger.info("NON_NIL_HEADERS -> #{headers}")
+    %{response | headers: non_nil_headers}
```

Comment:
The server already logs the response, which includes all the headers, so this is redundant logging (unless it was temporary to troubleshoot).  I suggest removing the logging, and keeping in pattern with the suggest above, update this piece of code as well.
```suggestion
    %{response | headers: collect_headers}
```

Reply:
Whoops! The Logger.info was left in from me trying to debug an issue I ran into with the integration tests. Thanks for calling it out!